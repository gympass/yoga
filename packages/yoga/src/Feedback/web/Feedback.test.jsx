import React from 'react';
import { render } from '@testing-library/react';

import { ThemeProvider, Feedback } from '../..';

function renderWithTheme(ui) {
  return render(ui, { wrapper: ThemeProvider });
}

const title = 'Welcome to Yoga';
const description = 'Enjoy your membership!';
const caption = 'Caption Description here.';

describe('<Feedback />', () => {
  it('should render the title and description', () => {
    const { getByText } = renderWithTheme(
      <Feedback variant="success" title={title} description={description} />,
    );

    expect(getByText(title)).toBeTruthy();
    expect(getByText(description)).toBeTruthy();
  });

  it('should render the title children', () => {
    const { getByRole } = renderWithTheme(
      <Feedback variant="success" description="Enjoy your membership!">
        <Feedback.Title as="h1">{title}</Feedback.Title>
      </Feedback>,
    );

    expect(getByRole('heading', { name: title })).toBeTruthy();
  });

  it('should render caption children', () => {
    const { getByText } = renderWithTheme(
      <Feedback variant="success" title={title} description={description}>
        <Feedback.Caption>{caption}</Feedback.Caption>
      </Feedback>,
    );

    expect(getByText(caption)).toBeTruthy();
  });

  it('should render the buttons', () => {
    const { getByRole } = renderWithTheme(
      <Feedback variant="success" title={title} description={description}>
        <Feedback.PrimaryButton>Ok</Feedback.PrimaryButton>
        <Feedback.SecondaryButton>Cancel</Feedback.SecondaryButton>
      </Feedback>,
    );

    expect(getByRole('button', { name: 'Ok' })).toBeTruthy();
    expect(getByRole('button', { name: 'Cancel' })).toBeTruthy();
  });

  it('should render title, description, buttons and caption', () => {
    const { getByText, getByRole } = renderWithTheme(
      <Feedback variant="success" description="Enjoy your membership!">
        <Feedback.Caption>{caption}</Feedback.Caption>
        <Feedback.PrimaryButton>Ok</Feedback.PrimaryButton>
        <Feedback.SecondaryButton>Cancel</Feedback.SecondaryButton>
      </Feedback>,
    );

    expect(getByText(caption)).toBeTruthy();
    expect(getByRole('button', { name: 'Ok' })).toBeTruthy();
    expect(getByRole('button', { name: 'Cancel' })).toBeTruthy();
  });

  it('should render correctly when not centered vertically', () => {
    const { baseElement } = renderWithTheme(
      <Feedback
        variant="success"
        title={title}
        description={description}
        center={false}
      />,
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('should render the icon for success variant', () => {
    const { getByTestId } = renderWithTheme(
      <Feedback variant="success" title={title} description={description} />,
    );

    expect(getByTestId('feedback-icon-success')).toMatchSnapshot();
  });

  it('should render the icon for informative variant', () => {
    const { getByTestId } = renderWithTheme(
      <Feedback
        variant="informative"
        title={title}
        description={description}
      />,
    );

    expect(getByTestId('feedback-icon-informative')).toMatchSnapshot();
  });

  it('should render the icon for attention variant', () => {
    const { getByTestId } = renderWithTheme(
      <Feedback variant="attention" title={title} description={description} />,
    );

    expect(getByTestId('feedback-icon-attention')).toMatchSnapshot();
  });

  it('should render the icon for delayed variant', () => {
    const { getByTestId } = renderWithTheme(
      <Feedback variant="delayed" title={title} description={description} />,
    );

    expect(getByTestId('feedback-icon-delayed')).toMatchSnapshot();
  });
});
