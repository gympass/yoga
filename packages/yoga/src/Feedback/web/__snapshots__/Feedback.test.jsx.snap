// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Feedback /> should render correctly when not centered vertically 1`] = `
.c2 {
  margin-bottom: 24px;
  width: 64px;
  height: 64px;
}

.c0 {
  background-color: #FFFFFF;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c5 {
  margin: 0;
  padding: 0;
  font-size: 16px;
  line-height: 24px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 400;
  margin-top: 16px;
  color: #6B6B78;
  font-size: 16px;
}

.c1 {
  width: 100%;
}

.c3 {
  max-width: 100%;
}

.c4 {
  margin: 0;
  padding: 0;
  font-size: 24px;
  line-height: 32px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 500;
}

.c6 {
  margin-top: 56px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 16px;
}

@media (min-width:769px) {
  .c1 {
    margin: auto;
  }
}

@media (min-width:769px) {
  .c3 {
    max-width: 520px;
  }
}

@media (min-width:769px) {
  .c4 {
    font-size: 48px;
    line-height: 48px;
  }
}

@media (min-width:769px) {
  .c6 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    margin-top: 40px;
    gap: 0;
  }
}

<body>
  <div>
    <div
      class="c0"
      display="flex"
      width="100%"
    >
      <div
        class="c1"
      >
        <svg
          aria-hidden="true"
          class="c2"
          data-testid="feedback-icon-success"
          height="64"
          mb="large"
          width="64"
        />
        <div
          class="c3"
        >
          <p
            class="c4"
          >
            Welcome to Yoga
          </p>
          <p
            class="c5"
            color="deep"
            font-size="medium"
          >
            Enjoy your membership!
          </p>
        </div>
        <div
          class="c6"
        />
      </div>
    </div>
  </div>
</body>
`;

exports[`<Feedback /> should render the icon for attention variant 1`] = `
.c0 {
  margin-bottom: 24px;
  width: 64px;
  height: 64px;
}

<svg
  aria-hidden="true"
  class="c0"
  data-testid="feedback-icon-attention"
  height="64"
  mb="large"
  width="64"
/>
`;

exports[`<Feedback /> should render the icon for delayed variant 1`] = `
.c0 {
  margin-bottom: 24px;
  width: 64px;
  height: 64px;
}

<svg
  aria-hidden="true"
  class="c0"
  data-testid="feedback-icon-delayed"
  height="64"
  mb="large"
  width="64"
/>
`;

exports[`<Feedback /> should render the icon for informative variant 1`] = `
.c0 {
  margin-bottom: 24px;
  width: 64px;
  height: 64px;
}

<svg
  aria-hidden="true"
  class="c0"
  data-testid="feedback-icon-informative"
  height="64"
  mb="large"
  width="64"
/>
`;

exports[`<Feedback /> should render the icon for success variant 1`] = `
.c0 {
  margin-bottom: 24px;
  width: 64px;
  height: 64px;
}

<svg
  aria-hidden="true"
  class="c0"
  data-testid="feedback-icon-success"
  height="64"
  mb="large"
  width="64"
/>
`;
